src/time.rs

```diff
@@ -155,3 +155,42 @@ impl TimeProvider for DefaultTimeProvider {
 }
 
 pub(crate) static DEFAULT_TIME_PROVIDER: DefaultTimeProvider = DefaultTimeProvider { _dummy: () };
+
+#[cfg(test)]
+mod tests {
+    use super::{Date, Time};
+
+    #[test]
+    fn date_encode_decode() {
+        let d = Date {
+            year: 2055,
+            month: 7,
+            day: 23
+        };
+        assert_eq!(d, Date::decode(d.encode()));
+    }
+
+    #[test]
+    fn time_encode_decode() {
+        let t1 = Time {
+            hour: 15,
+            min: 3,
+            sec: 29,
+            millis: 990,
+        };
+        let t2 = Time {
+            sec: 18,
+            .. t1
+        };
+        let t3 = Time {
+            millis: 40,
+            .. t1
+        };
+        let (x1, y1) = t1.encode();
+        let (x2, y2) = t2.encode();
+        let (x3, y3) = t3.encode();
+        assert_eq!(t1, Time::decode(x1, y1));
```

Comment:
Please add some assertions that checks returned xN and yN values (current assertions only check if encoding can be reversed using decode function, not if encoded values are actually correct)

Reply:
Done.